plutus-core/plutus-ir/src/PlutusIR/Compiler/Datatype.hs

```diff
@@ -189,10 +176,89 @@ forall (t_1 :: *) .. (t_n :: *) .
 
 The term for the destructor is:
 /\(t_1 :: *) .. (t_n :: *) .
-  \(x :: (T t_1 .. t_n)) ->
+  \(x :: (T t_1 .. t_n)) .
     x
+-}
+
+{- Note [SOP encoding of datatypes]
```

Comment:
The point is to have a note for each that just explains carefully how it works, since they're both quite fiddly. I'll add a third note that links them.